src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro128StarStarImpl.cs

```diff
@@ -28,6 +28,11 @@ internal sealed class XoshiroImpl : ImplBase
             //
             //     See <http://creativecommons.org/publicdomain/zero/1.0/>.
 
+            // Next(int maxValue), Next(int minValue, int maxValue), NextInt64(long maxValue) and NextInt64(int minValue, int maxValue)
+            // are based on the algorithm from https://arxiv.org/pdf/1805.10941.pdf:
+            //
+            //     Written in 2018 by Daniel Lemire
```

Comment:
could you please add a section in the THIRD-PARTY-NOTICES.TXT at the root as well.

Reply:
Done.